### PR TITLE
Resolve CS4014 and CA1001

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Common/Amqp/ClientWebSocketTransport.cs [hunk 1]
```diff
@@ -64,55 +64,6 @@ public override bool WriteAsync(TransportAsyncCallbackArgs args)
             return true;
         }
 
-        private async Task WriteAsyncCoreAsync(TransportAsyncCallbackArgs args)
```
Comment (drwill-ms) [inline iothub/device/src/Common/Amqp/ClientWebSocketTransport.cs:67]: I moved ordered methods by visibility scope

File: iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs [hunk 2]
```diff
@@ -62,81 +57,46 @@ public async Task<AmqpIoTConnection> OpenConnectionAsync(TimeSpan timeout)
...
         private static bool InitializeDisableServerCertificateValidation()
         {
 #if !NET451
-            bool flag;
-            if (!AppContext.TryGetSwitch("DisableServerCertificateValidationKeyName", out flag))
-            {
-                return false;
-            }
-            return flag;
+            return AppContext.TryGetSwitch("DisableServerCertificateValidationKeyName", out bool flag) && flag;
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs:89]: nice!

File: iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs [hunk 3]
```diff
@@ -62,81 +57,46 @@ public async Task<AmqpIoTConnection> OpenConnectionAsync(TimeSpan timeout)
...
-                }
+                Logging.Exit(this, timeout, $"{nameof(OpenConnectionAsync)}");
 
                 return amqpIoTConnection;
             }
-            catch (Exception e) when (!e.IsFatal())
+            catch (Exception ex) when (!ex.IsFatal())
             {
                 transportBase?.Close();
+                _amqpIotTransport?.Dispose();
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs:77]: _amqpIotTransport = null ?

File: iothub/device/tests/ClientWebSocketTransportTests.cs [hunk 4]
```diff
@@ -18,33 +18,38 @@ public class ClientWebSocketTransportTests
...
         private const string IotHubName = "localhost";
         private const int Port = 12345;
-        private static readonly Action<TransportAsyncCallbackArgs> onReadOperationComplete = OnReadOperationComplete;
-        private static readonly Action<TransportAsyncCallbackArgs> onWriteOperationComplete = OnWriteOperationComplete;
-        private static ClientWebSocketTransport clientWebSocketTransport;
-        private static Task s_serverTask;
 
-        private static byte[] byteArray = new byte[10] { 0x5, 0x6, 0x7, 0x8, 0xA, 0xB, 0xC, 0xD, 0xE, 0xF };
-        private static volatile bool readComplete;
+        private static readonly TimeSpan s_oneSecond = TimeSpan.FromSeconds(1);
```
Comment (abhipsaMisra) [inline iothub/device/tests/ClientWebSocketTransportTests.cs:22]: 😆

File: iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs (hunk 3 above)
Comment (drwill-ms) [reply iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs:77]: Why?

**Review by azabbasi — APPROVED** [review]

File: iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs (hunk 3 above)
Comment (drwill-ms) [reply iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs:77]: I'll fix this in my next PR. I understand now.

**Pull request merged**